### PR TITLE
Adding modals

### PR DESCRIPTION
**Pull request by LeviTrammell**
<details>
<summary>File System Diff</summary>

```diff
@@ -1,17 +1,20 @@
 .
   |-src
   |  |-domains
   |  |  |-core
   |  |  |  |-screens
+  |  |  |  |  |-RootStackNavigator.tsx
   |  |  |  |  |-BottomTabNavigator.tsx
   |  |  |  |-App.tsx
   |  |  |-todoApp
   |  |  |  |-screens
   |  |  |  |  |-SyncReduxTodoScreen.tsx
   |  |  |  |  |-App.tsx
+  |  |  |-modalTutorial
+  |  |  |  |-ModalTutorialScreen.tsx
   |  |  |-home
   |  |  |  |-HomeScreen.tsx
   |  |-state
   |  |  |-index.ts
   |  |  |-todoSlice.ts
   |-package.json
```
</details>

### CODE REVIEW

File: src/domains/core/screens/RootStackNavigator.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { ModalTutorialScreen } from 'src/domains/modalTutorial/ModalTutorialScreen';
+
+export type RootStackNavigatorParams = {
+  BottomTabNavigator: undefined;
+  ModalTutorialScreen: undefined;
+};
+
+const Stack = createStackNavigator<RootStackNavigatorParams>();
+
+export type RootStackNavigationProp<ScreenName extends keyof RootStackNavigatorParams> = StackNavigationProp<
```
Comment (LeviTrammell) [inline src/domains/core/screens/RootStackNavigator.tsx:13]: We can compose a few different types of these `NavigationProp` types to make it simpler to pull in.

File: src/domains/home/HomeScreen.tsx [hunk 2]
```diff
@@ -57,13 +58,14 @@ const Section: FC<{ title: string }> = ({ children, title }) => {
 
 export const HomeScreen: FC = () => {
   const isDarkMode = useColorScheme() === 'dark';
+  const navigation = useNavigation<RootStackNavigationProp<'BottomTabNavigator'>>();
```
Comment (garrettjoecox) [inline src/domains/home/HomeScreen.tsx]: Why is `BottomTabNavigator` passed in here?

**Review by garrettjoecox — APPROVED** [review]
1 question, aside from that LGTM

File: src/domains/home/HomeScreen.tsx (hunk 2 above)
Comment (LeviTrammell) [reply src/domains/home/HomeScreen.tsx]: Fixed this.

File: src/domains/home/HomeScreen.tsx [hunk 3]
```diff
@@ -57,13 +58,14 @@ const Section: FC<{ title: string }> = ({ children, title }) => {
 
 export const HomeScreen: FC = () => {
   const isDarkMode = useColorScheme() === 'dark';
+  const navigation = useNavigation<BottomTabNavigatorProp<'Home'>>();
```
Comment (garrettjoecox) [inline src/domains/home/HomeScreen.tsx:61]: 👍

**Review by garrettjoecox — APPROVED** [review]

**Pull request merged**